cmake/DownloadBoost.cmake

```diff
@@ -26,7 +26,7 @@ function(download_boost SOURCE_DIR BUILD_DIR BOOST_LIBS)
   set(BOOST_INSTALL_DIR "${BUILD_DIR}/dependencies/boost-install")
 
   set(CFLAGS "-fPIC")
-  set(CXXFLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+  set(CXXFLAGS "${CMAKE_CXX_FLAGS} -fPIC -std=c++14")
```

Comment:
These are not inherited from the root CMakeLists?

Reply:
In my case they were not; however, I started using an IDE that ran a cmake command and set this, so it overrode this. I didn't dive into why, but it called into question whether we should be using add_compile_option instead. I'm okay removing this and looking into that later. setting CMAKE_CXX_CLAGS is something I've been doing for a while but have had issues like this in various cases. I've found add_compile_option to be safer because then any subproject and subdirectory is guaranteed to have it as setting it doesn't force a cache when it's called.